test/strings/search.jl

```diff
@@ -65,6 +65,7 @@ end
 for str in (u8str, GenericString(u8str))
     @test_throws BoundsError findnext(equalto('z'), str, 0)
     @test_throws BoundsError findnext(equalto('∀'), str, 0)
+    @test findfirst('z', str) == nothing
```

Comment:
Could also test in a case that works, just in case? An easy way would be to loop over the predicate, using `identity` and `equalto`.